src/validation.cpp

```diff
@@ -1852,6 +1852,20 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // BIP148 mandatory segwit signalling.
+    int64_t nMedianTimePast = pindex->GetMedianTimePast();
+    if ( (nMedianTimePast >= 1501545600) &&  // Tue 01 Aug 2017 00:00:00 UTC
```

Comment:
Please take into consideration that - if the hashrate drops somehow, the activation period could take much more time (see the move of the acitvation period on the bitcoin UASF from november to august). So I would also suggest something like 1th June?